pkg/packaging/packaging.go

```diff
@@ -92,59 +92,54 @@ func (p *PackageOptions) Build(ctx context.Context, packageWriter io.Writer, tar
 		return errors.Wrap(err, "setup directories")
 	}
 
-	launcherEnv := map[string]string{
-		"KOLIDE_LAUNCHER_HOSTNAME":           p.Hostname,
-		"KOLIDE_LAUNCHER_ROOT_DIRECTORY":     p.rootDir,
-		"KOLIDE_LAUNCHER_OSQUERYD_PATH":      filepath.Join(p.binDir, "osqueryd"),
-		"KOLIDE_LAUNCHER_ENROLL_SECRET_PATH": filepath.Join(p.confDir, "secret"),
+	flagFilePath := filepath.Join(p.confDir, "launcher.flags")
+	flagFile, err := os.Create(filepath.Join(p.packageRoot, flagFilePath))
+	if err != nil {
+		return errors.Wrap(err, "creating flag file")
+	}
+	defer flagFile.Close()
+
+	launcherMapFlags := map[string]string{
+		"hostname":           p.Hostname,
+		"root_directory":     p.canonicalizePath(p.rootDir),
+		"osqueryd_path":      p.canonicalizePath(filepath.Join(p.binDir, "osqueryd")),
+		"enroll_secret_path": p.canonicalizePath(filepath.Join(p.confDir, "secret")),
 	}
 
-	launcherFlags := []string{}
+	launcherBoolFlags := []string{}
 
 	if p.InitialRunner {
-		launcherFlags = append(launcherFlags, "--with_initial_runner")
+		launcherBoolFlags = append(launcherBoolFlags, "with_initial_runner")
 	}
 
 	if p.Control && p.ControlHostname != "" {
-		launcherEnv["KOLIDE_LAUNCHER_CONTROL_HOSTNAME"] = p.ControlHostname
+		launcherMapFlags["control_hostname"] = p.ControlHostname
 	}
 
 	if p.Autoupdate && p.UpdateChannel != "" {
-		launcherFlags = append(launcherFlags, "--autoupdate")
```

Comment:
interesting, so if we're using a flagfile we just put a flag per line and skip the `-`/`--`?

Reply:
Empirically, yes. That seems to be how `ff` works. (it also supports json and yaml, but this was good enough)

Reply:
interesting choice by `ff` there. Thanks for the explanation